cardano-explorer-db/src/Explorer/DB/Query.hs

```diff
@@ -152,6 +158,27 @@ queryLatestSlotNo = do
             pure (blk ^. BlockSlotNo)
   pure $ fromMaybe 0 (listToMaybe $ mapMaybe unValue res)
 
+-- | Get the network metadata.
+queryMeta :: MonadIO m => ReaderT SqlBackend m (Either LookupFail Meta)
+queryMeta = do
+  res <- select . from $ \ (meta :: SqlExpr (Entity Meta)) -> do
+            pure meta
+  pure $ case res of
+            [] -> Left DbMetaEmpty
+            [m] -> Right $ entityVal m
+            _ -> Left DbMetaMultipleRows
+
+-- | Calculate the slot time for a given slot number.
+querySlotTime :: MonadIO m => Word64 -> ReaderT SqlBackend m (Either LookupFail UTCTime)
+querySlotTime slotNo = do
+  runExceptT $ do
+    meta <- ExceptT queryMeta
```

Comment:
would it be possible to cache the `queryMeta` somehow, i dont like the idea of fetching it for every single block, because we know the slot-duration/start-time cant change mid-chain

Reply:
This query was written as example. WOuld like to do it better, but have not really thought about how to do it better.

Reply:
I have added a comment there.